### DISCUSSION TITLE
Auto Generate Lightning Invoice from Receipt Capture

### DISCUSSION THREAD

**Discussion by ThinusXpressed** [btcpayserver/btcpayserver#5792] [Ideas / Feature Requests]
"I have a crazy idea that I need to share with you and find out what you think before I spend too much time writing a request for BTC Pay Server...

...maybe it already exists and I just don't know about it... but what if with BTC Pay Server's PoS, a receipt capture is implemented, then the software scans the receipt, checks the currency symbol and amount, and generates an invoice accordingly, thus no manual input and the image is stored with the note for consolidation purposes.

Digitally extracting the information like taking a picture of notes that gets converted into a .pdf."

The request is necessary because from my interaction and perspective most corporate structured organizations will retain receipts and utilize their existing Point of Sale (PoS) software for consolidation and balance sheet purposes...

@pavlenex @

**Comment by ThinusXpressed** [comment]
☝️@pavlenex , do you think this is a good idea and what would be the next steps...

**Reply by pavlenex** [reply]
Honestly, I am having very hard times understanding what is it you're requesting, what's the actual use-case and how it can be implemented, perhaps you should write a more structured proposal on the feature or provide specific examples.

**Reply by ThinusXpressed** [reply]
@pavlenex , thanks for the reply, let me apply my mind and draft the request in a more structure way, please allow me a few days to do this... Thanks!

**Reply by ThinusXpressed** [reply]
@pavlenex I realised that if I leave this for later it might not get done so I asked a bit of assistance from chat GPT is this should provide a better idea of what I think is possible.

**Objective:**
The primary objective of this feature enhancement is to simplify and automate the transaction process for merchants using the BTC Pay Server Point of Sale system, particularly in scenarios where traditional systems are still being used alongside lightning payments.

**Feature Description:**

**1. Open the BTC Pay Server Point of Sale:**
- The feature will be seamlessly integrated into the existing BTC Pay Server Point of Sale interface, ensuring a cohesive user experience.

**2. Image Capture Option:**
- After a traditional [fiat based receipt] is printed, users will have the option to capture an image of the receipt directly within the BTC Pay Server Point of Sale interface.
- This functionality can be accessed through a designated button or menu option, easily accessible to users.

**3. Receipt Image Analysis:**
- Upon capturing the receipt image, the software will initiate an analysis process to extract key transaction details, including merchant information, transaction date, amount, and currency.
- The system will identify the currency symbol to determine the currency used in the transaction.

**4. Automatic Lightning Invoice Generation:**
- Based on the extracted transaction details, including the currency and amount, the software will automatically generate a lightning invoice denominated in SATS, using the current exchange rate.
- This eliminates the need for manual invoice generation by the merchant, streamlining the transaction process.

**5. Embedding Receipt in Lightning Transaction Note:**
- As part of the lightning invoice generation process, the receipt captured as an image will be embedded within the lightning transaction note.
- This ensures that the receipt is securely associated with the corresponding transaction, facilitating easy reference and consolidation during daily accounting.

**The reason for this request** comes from observing the process followed by most merchants [especially coffe café's and reastuarants] accepting lightning use their traditional systems, print a receipt, and then the waiter have to manualy generate a lightning invoice, then the supervisor or manager must try and consolidate the traditional system with the payments received via the BTC Pay Server PoS creating double work and potentials for mistakes.

**Reply by pavlenex** [reply]
Interesting, so basically in invoices you want the ability to scan a receipt and generate an invoice and a receipt in btcpay based or it, or just the invoice? I just find it hard to think that this is easier than using btcpay pos and just typing the amount, steps you outlined is multi-step process requiring merchants to access the backend, with POS they just generate amount without even having to be logged in.

Probably more suitable for an external plugin if someone can build it.
